src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts

```diff
@@ -0,0 +1,112 @@
+import {ReadStillImageFileValue} from "../../../model/webapi/knora/v2/read-property-item";
+
+// OpenSeadragon does not export itself as ES6/ECMA2015 module,
+// it is loaded globally in scripts tag of angular-cli.json,
+// we still need to declare the namespace to make TypeScript compiler happy.
+declare var OpenSeadragon: any;
+
+/**
+* This component creates a OpenSeadragon viewer instance.
+* Accepts an array of ReadStillImageFileValues to be rendered.
+* @class
+* @member {ReadStillImageFileValue[]} fileValues - fileValues to be rendered. (Use as angular @Input data binding property.)
+* @member {boolean} ignorePreviews - controls if entries in fileValues with isPreview==true should be ignored or not. (Use as angular @Input data binding property.)
+*/
+@Component({
+  selector: 'salsah-still-image-osdviewer',
+  templateUrl: './still-image-osdviewer.component.html',
+  styleUrls: ['./still-image-osdviewer.component.scss']
+})
+export class StillImageOSDViewerComponent implements OnInit {
+
+    @Input() fileValues: ReadStillImageFileValue[] = [];
+    @Input() ignorePreviews: boolean = true;
+
+    private viewer;
+
+    constructor(private elementRef: ElementRef) { }
+
+    ngOnInit() {
+      // initialisation is done on first run of ngOnChanges
+    }
+
+    ngOnChanges(changes:{[key: string]: SimpleChange}) {
+      if (changes["fileValues"].isFirstChange()) {
+        this.setupViewer();
+      }
+      this.openImages();
+    }
+
+    ngOnDestroy() {
+      if (this.viewer) {
+        this.viewer.destroy();
+        this.viewer = null;
+      }
+    }
+
+    private setupViewer() {
+      let viewerContainer = this.elementRef.nativeElement.getElementsByClassName("osdViewerContainer")[0];
+      let osdOptions = {
+        element: viewerContainer,
+        prefixUrl: "assets/icons/openseadragon/"
+      }
+      this.viewer = new OpenSeadragon.Viewer(osdOptions);
+      this.viewer.addHandler('full-screen', function (args) {
+        if (args.fullScreen) {
+          viewerContainer.classList.add("fullscreen");
+        } else {
+          viewerContainer.classList.remove("fullscreen");
+        }
+      })
+    }
+
+    private openImages() {
+      let imagesToDisplay = this.getImagesToDisplay();
+      let tileSources = [];
+      for (let i = 0; i < imagesToDisplay.length; i++) {
+        let sipiBasePath = imagesToDisplay[i].imageServerIIIFBaseURL + "/" + imagesToDisplay[i].imageFilename;
+        let width = imagesToDisplay[i].dimX;
+        let height = imagesToDisplay[i].dimY;
+        tileSources[i] = {
+          "tileSource": {
+           "@context": "http://iiif.io/api/image/2/context.json",
+           "@id": sipiBasePath,
+            "height": height,
+            "width": width,
+            "profile": [ "http://iiif.io/api/image/2/level2.json" ],
+            "protocol": "http://iiif.io/api/image",
+            "tiles": [{
+              "scaleFactors": [ 1, 2, 4, 8, 16, 32 ],
+              "width": 1024
```

Comment:
see:
http://iiif.io/api/image/2.1/#technical-properties 
https://openseadragon.github.io/docs/OpenSeadragon.Viewer.html#open

the wording in the first link is a bit ambiguous, more detailed info in the implementation notes:
http://iiif.io/api/image/2.0/#a-implementation-notes
or directly in the sources:
https://github.com/openseadragon/openseadragon/blob/cfc339edea0a65575c7904ec82e72e911c584825/src/iiiftilesource.js